### ISSUE TITLE
Explicit keyboard binding when clause context for "Terminal in editor area"

### ISSUE THREAD

**Issue by CombeeMike** [microsoft/vscode#129672]
Is there an explicit keyboard binding when clause context to assign different key bindings to focused integrated & "in editor area" terminals?

If not, I'd really like to see them added. Something like `terminalEditorFocus` in addition to the already existing `terminalFocus`.

# Example use case

I have assigned `Ctrl+F4` to `workbench.action.terminal.kill` when `terminalFocus` and to `workbench.action.closeActiveEditor` as a fallback (when no terminal has focus).

However, the `terminalFocus` when clause is also true when an "in editor area terminal" has focus for which the `workbench.action.terminal.kill` doesn't work since a dedicated `workbench.action.terminal.killEditor` exists.

In other words, I'd need the following 3 key bindings:
* `Ctrl+F4` -> `workbench.action.terminal.kill` when `terminalFocus`
* `Ctrl+F4` -> `workbench.action.terminal.killEditor` when `terminalEditorFocus` (not possible due to missing when context)
* `Ctrl+F4` -> `workbench.action.closeActiveEditor` when `-`

**Comment by meganrogge** [comment]
I believe what you're looking for was fixed with https://github.com/microsoft/vscode/issues/128354

**Comment by CombeeMike** [comment]
@meganrogge

I'm a little doubtful of whether the issue mentioned by you actually solves my issue or not.
Can you tell me if the fix mentioned there will be included in the next VSCode release? If so, I simply wait until the next release and re-try...

**Comment by meganrogge** [comment]
can you try in the insider's build? it's fixed there

**Comment by CombeeMike** [comment]
@meganrogge 

I just tried the latest `v1.59.0-insider` and I don't see how my issue is fixed there.

There are still 2 explicit commands `workbench.action.terminal.kill` and `workbench.action.terminal.killEditor` which either work on the integrated or the "in editor" terminal but none of them is working on both.

The command `workbench.action.closeActiveEditor` works on both, a "normal" editor and the "in editor terminal", however, I still don't see no way of assigning keybindings to `Ctrl + F4` which work on all 3 of them ("normal" editor, integrated terminal & in editor terminal).

Could you explain a little more in detail how the changes in the latest build could help me here?

**Comment by GongT** [comment]
Please reopen this issue. #128354 is not same with this.

I need something like `activeEditorIsTerminal` or `terminalEditorFocus`. My problem is `workbench.action.terminal.focusNext` is conflict with `workbench.action.nextEditorInGroup`.

![动画](https://user-images.githubusercontent.com/2265048/128465939-776b3d67-5853-4fe1-976d-2df472a659bf.gif)

**Comment by Tyriar** [comment]
We have `terminalEditorFocus` now:

https://github.com/microsoft/vscode/blob/a26f3b00c0039cce6164b0c78b19b284a273c189/src/vs/workbench/contrib/terminal/common/terminalContextKey.ts#L37-L38

![image](https://user-images.githubusercontent.com/2193314/128713964-2fc37163-8aa9-4f2e-8505-7b3738e8d30e.png)

But insiders has ctrl+f4 working for terminal editors already, to get it working in the panel you should also add this keybinding:

```
{
	"key": "ctrl+f4", "command": "workbench.action.terminal.kill", "when": "terminalFocus"
}
```
